### ISSUE TITLE
~ tab completion issues on NFS-Home with Kerberos-User

### ISSUE THREAD

**Issue by fkjogu** [fish-shell/fish-shell#2153]
I'm using fish 2.1.2 on Debian Jessie.

Wenn typing
```
cd ~/some/dir
```
everything works fine. But I'd expect that
```
cd ~/so<tab>
```
would complete _~/so_ to to _~/some_. Instead nothing happens. This is also the case when using the absolute home dir path
```
cd /absolute/path/to/my/home/dir/so<tab> # no completion
```

Other tab completion features seem to work, including relative path completion inside the home directory.

I'm not sure if it is relevant, but the user is part of a Kerberos infrastructure and the home directory is mounted  on a NFS device.

**Comment by zanchey** [comment]
One problem we've seen a few times recently is described (and fixed) in #2099, where some components of the path are not readable to the current user.

**Comment by fkjogu** [comment]
Just tested master and can confirm fix by #2099. Wonderful, closing.
